src/zenml/cli/stack_components.py

```diff
@@ -235,22 +236,126 @@ def register_stack_component_command(
+            component_type=component_type
+        )
+
+        try:
+            # Try to find if there are any custom flavor implementations
+            flavor_wrapper = Repository().zen_store.get_flavor_by_name_and_type(
+                flavor_name=flavor,
+                component_type=component_type,
             )
-            component = component_class(name=name, **parsed_args)
+
+            # If there is one, check whether the same flavor exists as a default
+            # flavor to give out a warning
+            if flavor in zenml_flavors:
+                cli_utils.warning(
+                    f"There is a custom implementation for the flavor "
+                    f"'{flavor}' of a {component_type}, which is currently "
+                    f"overwriting the same flavor provided by ZenML."
+                )
+
+        except KeyError:
+            if flavor in zenml_flavors:
+                flavor_wrapper = zenml_flavors[flavor]
+            else:
+                raise KeyError(
+                    f"There is no flavor '{flavor}' for the type "
+                    f"{component_type}"
+                )
+
+        try:
+            component = flavor_wrapper.to_flavor()(name=name, **parsed_args)
             Repository().register_stack_component(component)
             cli_utils.declare(
                 f"Successfully registered {display_name} `{name}`."
             )
 
+        except (ModuleNotFoundError, ImportError, NotImplementedError):
+            if flavor_wrapper.integration:
+                cli_utils.error(
+                    f"The {component_type} flavor '{flavor_wrapper.name}' is "
+                    f"a part of ZenML's '{flavor_wrapper.integration}' "
+                    f"integration, which is currently not installed on your "
+                    f"system. You can install it by executing: 'zenml "
+                    f"integration install {flavor_wrapper.integration}'."
+                )
+            else:
+                cli_utils.error(
+                    f"The custom flavor '{flavor_wrapper.name}' can not be "
+                    f"used as it can not be imported in its current state. "
+                    f"Please make sure that the implementation is in a state "
+                    f"which can be imported as a module."
+                )
+
     return register_stack_component_command
 
 
+def generate_stack_component_flavor_register_command(
+    component_type: StackComponentType,
+) -> Callable[[str], None]:
+    """Generates a `register` command for the flavors of a stack component."""
+
+    @click.argument(
+        "source",
+        type=str,
+        required=True,
+    )
+    def register_stack_component_flavor_command(source: str) -> None:
+        """Adds a flavor for a stack component type"""
+        cli_utils.print_active_profile()
+
+        # Check whether the module exists and is the right type
+        component_class = cli_utils.validate_flavor_source(
```

Comment:
Should we catch the errors here and try to give a better error message instead of the `ImportError`?

Reply:
Updated the validation function.